### ISSUE TITLE
Journal: Flagged legs aren't highlighted

### ISSUE THREAD

**Issue by corani** [aumayr/fava#195]
You can also attach flags to the postings themselves, if you want to flag one of the transaction’s legs in particular:
```
2014-05-05 * "Transfer from Savings account"
  Assets:MyBank:Checking            -400.00 USD
  ! Assets:MyBank:Savings
```

Ideal:
When the transaction is collapsed, the entire transaction should be highlighted (as if you'd flagged the transaction with `!`). When it's uncollapsed, the flagged leg should be highlighted in the same way.

Acceptable:
When a leg of a transaction is flagged, the entire transaction should be highlighted and the flagged leg should be highlighted differently.

**Comment by yagebu** [comment]
> When the transaction is collapsed, the entire transaction should be highlighted (as if you'd flagged the transaction with !).

I don't agree. The whole transaction already has a flag (`*`), so the highlighting of the transaction should always derive from that. If you want to flag the whole transaction as pending, then `!` the whole transaction.

But we should highlight the posting according to its flag if it's different from the transaction flag.

**Comment by corani** [comment]
> The whole transaction already has a flag (*), so the highlighting of the transaction should always derive from that.

The issue is that when the transaction has a `*` flag, when legs are collapsed, it's impossible to find those transactions with flagged legs.

**Comment by yagebu** [comment]
> The issue is that when the transaction has a * flag, when legs are collapsed, it's impossible to find those transactions with flagged legs.

And that's sensible behaviour. If you want information about the postings you should display them.

**Comment by corani** [comment]
I don't agree. If one of the postings is flagged, obviously there's still something wrong with the entire transaction. 

(by the way, even collapsed my page is already about 100,000 pixels high. Displaying the postings makes it more than twice that. Try finding anything in that...)

**Comment by yagebu** [comment]
> I don't agree. If one of the postings is flagged, obviously there's still something wrong with the entire transaction.

Why should an arbitrary posting flag overwrite the transaction flag? Flags don't have any inherent semantic meaning and their meaning is for the user to define. Just because for you `!` means wrong, doesn't mean everyone uses it in the same way. Also if there's something "wrong" with the whole transaction, why not just flag the whole transaction with `!`? (this could be done with a plugin that changes the transaction flag if one of the postings is flagged with `!`).

**Comment by aumayr** [comment]
IMHO you are both right:

- If a leg is flagged, the leg should be highlighted accordingly.
- If the transaction is collapsed, and just a leg is flagged, it should not look the same as if the transaction itself would be flagged.
- If the transaction is collapsed, and just a leg is flagged, it should be somehow visible to the user that there is a flagged leg. 

So what about adding an indicator or something to inform the user, if transactions are collapsed, that there is a leg with a flag?

**Comment by yagebu** [comment]
Indicators sound like a good idea (I imagine one colored dot for each posting). This will also make it easier to spot transactions with lots of postings.

**Comment by aumayr** [comment]
> I imagine one colored dot for each posting

One colored dot for each posting with a flag you mean?

**Comment by yagebu** [comment]
> One colored dot for each posting with a flag you mean?

I actually meant one for each posting (so if it has no flag, the dot would have the same color as the transaction, maybe with a dark border to make it visible). This way it would serve two purposes: 1. Show flagged postings and 2. show number of postings. The first one is not interesting for me while the second one might be.

**Comment by aumayr** [comment]
> I actually meant one for each posting

Sounds good to me. I will have a go at it.

**Comment by aumayr** [comment]
I implemented the dots in d84d78b, and I'm quite happy with them. 

What do you think, @corani and @yagebu ?

**Comment by corani** [comment]
Looks great!
